BLSQ/iaso
# POLIO-1769: warn users when switching scope type

Users need to be warned that changing the scope level will delete the old existing scopes

Related JIRA tickets : POLIO-1769

## Self proofreading checklist


## Doc
N/a

## Changes

- Add WarningModal component
- Use it in polio modal when user saves and scope type has changed

## How to test

Explain how to test your PR.

Open a campaign, tick the "Separate scopes per round" box, add new scope, click confirm --> there should be a warning modal

## Print screen / video

![Screenshot 2024-11-29 at 16 35 27](https://github.com/user-attachments/assets/59cd0615-d2d5-41a4-86fb-c989c64044ed)


## Notes

MERGES INTO https://github.com/BLSQ/iaso/pull/1837

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: IA-3665
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.
